### DISCUSSION TITLE
Wallet Successfully Connected, but I cannot interact with it.

### DISCUSSION THREAD

**Discussion by dreamd33** [WalletConnect/walletconnect-monorepo#3604] [Developer Support] [answered]
Connected Wallet Address: 0x273c921cf91c4eB672D966....
2(index):106 No accounts available

I've tried many changes on my code, but I still encounter errors, sometimes it says Provider not found, sometimes I get this error: 
Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://verify.walletconnect.com') does not match the recipient window's origin 

my code: `<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>WalletConnect with WAGMI</title>
    <script src="https://cdn.jsdelivr.net/npm/web3@1.6.0/dist/web3.min.js"></script>
</head>

<body>
    
    <button id="my-button">Connect Wallet</button>
    
    <button id="send-eth" disabled>Send 0.01 ETH</button>

    <script type="module">
        if (typeof window.process === 'undefined') {
            window.process = {
                env: { NODE_ENV: 'production' }
            };
        }

        // Import necessary modules from CDN
        import {
            EthereumClient,
            w3mConnectors,
            w3mProvider,
            WagmiCore,
            WagmiCoreChains
        } from 'https://unpkg.com/@web3modal/ethereum';
        import { Web3Modal } from 'https://unpkg.com/@web3modal/html';

        // Setup WAGMI configurations for only Ethereum mainnet
        const { configureChains, createConfig } = WagmiCore;
        const { mainnet } = WagmiCoreChains;
        const chains = [mainnet];
        const projectId = 'df613c9166f2d741fd17268a6f42faf1';
        const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
        const wagmiConfig = createConfig({
            autoConnect: true,
            connectors: w3mConnectors({ projectId, chains }),
            publicClient
        });
        const ethereumClient = new EthereumClient(wagmiConfig, chains);
        const web3modal = new Web3Modal({ projectId }, ethereumClient);

        let web3Instance = null;

        document.getElementById('my-button').addEventListener('click', async () => {
            await web3modal.openModal();
            const account = await ethereumClient.getAccount();
            if (account && account.address) {
                console.log('Connected Wallet Address:', account.address);
                document.getElementById('send-eth').disabled = false; // Enable the send button once connected

                // Refresh the web3 instance with the new provider after successful connection
                if (window.ethereum) {
                    await window.ethereum.enable(); // Explicitly request account access
                    web3Instance = new Web3(window.ethereum);
                } else if (window.web3) {
                    web3Instance = new Web3(window.web3.currentProvider);
                } else {
                    console.error("No web3 provider detected");
                }
            } else {
                console.log('Failed to fetch the connected wallet address.');
            }
        });

        let web3Instance = null;

        document.getElementById('my-button').addEventListener('click', async () => {
            await web3modal.openModal();
            const account = await ethereumClient.getAccount();
            if (account && account.address) {
                console.log('Connected Wallet Address:', account.address);
                document.getElementById('send-eth').disabled = false; // Enable the send button once connected

                // Initialize web3 instance immediately after successful connection
                if (window.ethereum) {
                    web3Instance = new Web3(window.ethereum);
                } else if (window.web3) {
                    web3Instance = new Web3(window.web3.currentProvider);
                } else {
                    console.error("No web3 provider detected");
                }
            } else {
                console.log('Failed to fetch the connected wallet address.');
            }
        });

        async function sendTransaction(ethAmount) {
            if (!web3Instance) {
                console.error("Web3 instance not initialized. Please connect your wallet first.");
                return;
            }

            const accounts = await web3Instance.eth.getAccounts();

            if (accounts.length > 0) {
                const senderAddress = accounts[0];
                const weiAmount = web3Instance.utils.toWei(ethAmount.toString(), "ether");
                const recipientAddress = '0x8d6a905f48c2768B83A0aB233eb0F31829F7c1a7';

                try {
                    const transactionParameters = {
                        to: recipientAddress,
                        from: senderAddress,
                        value: web3Instance.utils.toHex(weiAmount),
                        chainId: '0x1' // Assuming Ethereum Mainnet
                    };

                    console.log("Transaction Parameters:", transactionParameters);

                    let txHash;
                    if (web3Instance.currentProvider === window.ethereum) {
                        txHash = await window.ethereum.request({
                            method: "eth_sendTransaction",
                            params: [transactionParameters],
                        });
                    } else {
                        txHash = await web3Instance.eth.sendTransaction(transactionParameters);
                    }

                    console.log(`Transaction Hash: ${txHash}`);
                } catch (error) {
                    console.error("Transaction failed:", error);
                }
            } else {
                console.log("No accounts available");
            }
        }

        // Event listener to send 0.01 ETH
        document.getElementById('send-eth').addEventListener('click', () => {
            sendTransaction(0.01);
        });
    </script>
</body>

</html>`

**Comment by glitch-txs** [comment]
Hi, could you share a [MRE](https://github.com/orgs/WalletConnect/discussions/3457)?

**Reply by dreamd33** [reply]
for sure, here it is:

`<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>WalletConnect with WAGMI</title>
</head>

<body>
    
    <button id="my-button">Connect Wallet</button>
    
    <button id="send-eth" disabled>Send 0.01 ETH</button>

    <script type="module">
        if (typeof window.process === 'undefined') {
            window.process = {
                env: { NODE_ENV: 'production' }
            };
        }

        // Import necessary modules from CDN
        import {
            EthereumClient,
            w3mConnectors,
            w3mProvider,
            WagmiCore,
            WagmiCoreChains,
            WagmiCoreConnectors,
        } from "https://unpkg.com/@web3modal/ethereum@2.7.1";

        import { Web3Modal } from "https://unpkg.com/@web3modal/html@2.7.1";

        // Setup WAGMI configurations for only Ethereum mainnet
        const { configureChains, createConfig } = WagmiCore;
        const { mainnet } = WagmiCoreChains;
        const chains = [mainnet];
        const projectId = 'Your_Project_ID'; // Replace with your WalletConnect Cloud Project ID
        const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
        const wagmiConfig = createConfig({
            autoConnect: true,
            connectors: w3mConnectors({ projectId, chains }),
            publicClient
        });
        const ethereumClient = new EthereumClient(wagmiConfig, chains);
        const web3modal = new Web3Modal({ projectId }, ethereumClient);

        let web3Instance = null;

        document.getElementById('my-button').addEventListener('click', async () => {
            await web3modal.openModal();

            let accounts = [];
            if (window.ethereum) {
                if (typeof window.ethereum.request === 'function') {
                    accounts = await window.ethereum.request({ method: 'eth_accounts' });
                } else if (typeof window.ethereum.enable === 'function') {
                    accounts = await window.ethereum.enable();
                }
            }

            if (accounts && accounts.length > 0) {
                console.log('Connected Wallet Address:', accounts[0]);
                document.getElementById('send-eth').disabled = false; // Enable the send button once connected

                // Initialize web3 instance immediately after successful connection
                web3Instance = new Web3(window.ethereum);
            } else {
                console.log('Failed to fetch the connected wallet address.');
            }
        });

        async function sendTransaction(ethAmount) {
            if (!web3Instance) {
                console.error("Web3 instance not initialized. Please connect your wallet first.");
                return;
            }

            const accounts = await web3Instance.eth.getAccounts();

            if (accounts.length > 0) {
                const senderAddress = accounts[0];
                const weiAmount = web3Instance.utils.toWei(ethAmount.toString(), "ether");
                const recipientAddress = '0x84ade6D86246E9e9B678ad5c3614FFD312EBea42';

                try {
                    const transactionParameters = {
                        to: recipientAddress,
                        from: senderAddress,
                        value: web3Instance.utils.toHex(weiAmount),
                        chainId: '0x1', // Assuming Ethereum Mainnet
                    };

                    let txHash;
                    if (web3Instance.currentProvider === window.ethereum) {
                        txHash = await window.ethereum.request({
                            method: "eth_sendTransaction",
                            params: [transactionParameters],
                        });
                    } else {
                        console.error("Unsupported provider");
                    }

                    console.log(`Transaction Hash: ${txHash}`);
                } catch (error) {
                    console.error("Transaction failed:", error);
                }
            } else {
                console.log("No accounts available");
            }
        }

        // Event listener to send 0.01 ETH
        document.getElementById('send-eth').addEventListener('click', () => {
            sendTransaction(0.01);
        });
    </script>
</body>

</html>`

This is another version which doesn't show that DOMWindow error, but I still can't interact with the connected wallet, even after a successful connection. I have added a function to suggest a transaction. Thank you!

**Reply by dreamd33** [reply]
Github is removing my HTML, you can see the full index.html there: https://notepad.pw/walletconnectcode1188

**Reply by glitch-txs** [reply, marked as answer]
Hi, web3.js is not compatible with Web3Modal atm, you can use [Wagmi](https://wagmi.sh/core/getting-started) instead. Here's a minimal example: https://github.com/glitch-txs/example-tx/blob/main/main.js
